### ISSUE TITLE
APK v1.0.0.2 is a test/ dev version (com.github.catfriend1.syncthingandroid_v1.0.0.2_ae2e1874.apk)

### ISSUE THREAD

**Issue by o-l-a-v** [Catfriend1/syncthing-android#205]
Seems like your most recent APK release on GitHub is a dev/ test version, which does not replace the v1.0.0.1 already installed on a device. Google Play update on my phone worked right, but manual APK update on my Shields did not. Different compiles?

[Screenshot from Solid Explorer](https://user-images.githubusercontent.com/6450056/50735541-1d4d4e80-11b1-11e9-8259-8c0d679c7467.jpg)

**Comment by Catfriend1** [comment]
Update works correctly if you're using the file from the same release type. If started with GitHub for example you'll need the github release apk to update. The same applies for Google play releases.
If it doesn't work, it may be Android refusing to update if the version number of the package didn't change.
I'll try it myself later. The apk has two compile flavours (debug, release) and three signatures (test, gplay release, fdroid release). If something went wrong, I'll recompile the github release and upload it again.
In the meantime, a workaround is available: export settings > uninstall > install update > import settings.

**Comment by Catfriend1** [comment]
**TL;DR:
If you use the GitHub release, stay on that "channel" and use the same release type for upgrading to a future version.
If you use the F-Droid release, stay on that "channel" and use F-Droid for upgrading to a future version.
If you use the Google Play release, stay on that "channel" and use Google Play Store for upgrading to a future version.**

Okay, digging deeper:
I found out that the GitHub releases are correctly signed for the "github channel" (test signature). The hash is "2ScaPj41giu4vFh+Y7Q0GJTqwbA=".

F-Droid uses their own certificates to guarantee the APK was built from the original source and repo stated on the release page. The hash is "nyupq9aU0x6yK8RHaPra5GbTqQY=".

The hash of the Google Play release is "dQAnHXvlh80yJgrQUCo6LAg4294=". That comes from release signing. So in my opinion, there's nothing to fix here, just to clarify.

**Comment by Catfriend1** [comment]
Here's a new wiki article about it:
https://github.com/Catfriend1/syncthing-android/blob/master/wiki/Switch-between-releases_Verify-APK-is-genuine.md
